### PR TITLE
Fix operator tests

### PR DESCRIPTION
**Pull request by matthiasgeihs**

### CODE REVIEW

File: eth/client.go [hunk 1]
```diff
@@ -155,6 +166,62 @@ func (cl *Client) SubscribeToBlocks() (*BlockSubscription, error) {
...
+
+	headerSub, err := cl.SubscribeNewHead(ctx, headers)
+	if err != nil {
+		return nil, fmt.Errorf("subscribing to blockchain head: %w", err)
+	}
+
+	blockSub := &BlockSubscription2{headerSub, blocks, errChan, startBlockNumber}
+
+	errg := errors.NewGatherer()
+	errg.Go(func() error {
```
Comment (sebastianst) [inline eth/client.go]: I wouldn't use an error gatherer to start a single routine and then start another routine just to write the result in the channel. you could do here e.g.
```suggestion
	subFn := func() error { // function body as it is }
	go func() {
		errChan <- subFn()
	}()
```

Comment (matthiasgeihs) [reply eth/client.go]: should be
```
go func() {
	err := subFn()
	errChan <- err
}()
```

ok, done

File: eth/client.go [hunk 2]
```diff
@@ -155,6 +165,62 @@ func (cl *Client) SubscribeToBlocks() (*BlockSubscription, error) {
...
+	}
+
+	blockSub := &BlockSubscription2{headerSub, blocks, errChan, startBlockNumber}
+
+	run := func() error {
+		for {
+			select {
+			case err := <-headerSub.Err():
+				if err != nil {
+					errChan <- fmt.Errorf("subscription error: %w", err)
```
Comment (sebastianst) [inline eth/client.go]: you write into the err chan here and then return the error, which will write it again?

Comment (matthiasgeihs) [reply eth/client.go]: nice catch

**Pull request merged**